### ISSUE TITLE
Set compile options in browser script attributes

### ISSUE THREAD

**Issue by johnjbarton** [google/traceur-compiler#1649]
eg
```html
<script type="module" traceur-options="--async-functions">
```

**Comment by arv** [comment]
It looks kind of ugly to use command line style in there but I could not think of a better way do it atm.

**Comment by UltCombo** [comment]
JS object syntax maybe?

```html
<script type="module" traceur-options="{ asyncFunctions: true }">
```

**Comment by arv** [comment]
That is better. Maybe even skip the curlies?

**Comment by UltCombo** [comment]
I agree curlies are a bit redundant, but the syntax may look a bit alien without them. Up to you guys.

**Comment by domenic** [comment]
I'd go for either skip the curlies or full-on-JSON (double quotes required). Basically, pick one of: exact match with usual configuration syntax, or user-friendliness.

**Comment by UltCombo** [comment]
full-on-JSON SGTM.

**Comment by johnjbarton** [comment]
I guess this has to block my options work or browsers can't set options.

Full on JSON seems human-unfriendly to me.

**Comment by johnjbarton** [comment]
To avoid options per module, the options should be on the traceur.js or bootstrap.js element.

**Comment by johnjbarton** [comment]
For the time being the workaround is to use
```js
$traceurRuntime.options = traceur.util.Options.experimental();
```

**Comment by sbrowngh** [comment]
To set experimental features,  

    traceur.options.experimental = true;

no longer seems to works for in-browser on-the-fly compilation using curent version of 

     https://google.github.io/traceur-compiler/bin/traceur.js
     https://google.github.io/traceur-compiler/src/bootstrap.js

Instead something like one of the following works:

     $traceurRuntime.options = traceur.util.Options.experimental();

     $traceurRuntime.options = new traceur.util.Options({....});

See discussion here: https://groups.google.com/d/msg/traceur-compiler-discuss/7HT3t9rrQkw/NTd4hLgeboUJ

**Comment by pflannery** [comment]
If its going be one monolithic attribute then JSON would be better from a programming point of view.

But I think for it to be more human friendly then it should be split in to multiple attributes similar to:

```html
<script type="text/traceur" async-functions array-comprehension async-generators>
...
</script>
```

At least then we can multi line the options when they grow too large.

**Comment by UltCombo** [comment]
@pflannery would be better to use data-attributes then for the sake of validness and to avoid conflicts with actual attributes.

**Comment by pflannery** [comment]
@UltCombo

That would look like:
```html
<script type="text/traceur" data-async-functions data-array-comprehension data-async-generators>
...
</script>
```

I think if any prefix it should be ```traceur-```

**Comment by pflannery** [comment]
Other possibilities to go with attributes could be:

- Registering a custom element called ```<traceur>``` and even ```<module>``` (when not present) , use the ```<traceur>``` for people that want options only bound to traceur and have ```<module>``` for expected native options. [shim for document.registerElement is here for non-native](https://github.com/webcomponents/webcomponentsjs/blob/e49de932c97ec336c6e1b106319a582e07d7385d/src/CustomElements/register.js#L68-L121)
- Parsing first line of the script for option comments like the REPL ```// Options: --async-functions```

**Comment by UltCombo** [comment]
@pflannery My comment was not about "choosing a prefix", but rather making the HTML valid.

`<script traceur-async-generators></script>` will not pass a simple [validation check](http://validator.w3.org/), but `<script data-async-generators></script>` will.

That is, I'd prefer to avoid promoting invalid HTML for no good reason. There are developers whose work requires such validity requirements, and promoting invalid markup would render that Traceur syntax unusable for them.

By the way, custom elements require a dash (`-`) in the name to be valid IIRC.

**Comment by pflannery** [comment]
@UltCombo oops yeah my bad, custom elements do need a (```-```). ```<traceur-script>``` would still be cool
Your point is valid regarding W3C but I use custom components heavily these days and they all have attributes defined in a manner that suits the element and not a standard. I used to be a developer who swore by W3C validity but very rarely use it these days (if at all). In the emerging world of libraries like Polymer a W3C validation check isn't useful until is supports such notions.

It makes me wonder, if a person is swearing by W3C standards and validation tools then should will they really be using a JS-next library anyway?

**Comment by johnjbarton** [comment]
Many developers using web pages can rely on one of two other solutions:
  1. precompilation and loading with `<script>` tags,
  2. `<script>` tags calling `System.import()`.

I think these solutions are adequate and very likely to work going forward.  Anything like a `<module>` tag is speculation and, IMO, not a good technical solution. So I'm not convinced solving this issue is worth much effort.

**Comment by UltCombo** [comment]
@pflannery if using a custom element (e.g. `<traceur-script>`), then attribute names are not a problem of course.

I particularly don't care so much about strictly following validators, but a future customer may care.

> It makes me wonder, if a person is swearing by W3C standards and validation tools then should will they really be using a JS-next library anyway?

The browser is executing compiled code, that is, valid ES5. From a validator's point of view, I believe there's no problem. The same holds true for any HTML/CSS/JS pre-processor.

**Comment by johnjbarton** [comment]
Since it was easy, I implemented using the command line options in PR #2103.
